test/integration/registration/spokecluster_aws_joining_test.go

```diff
@@ -75,37 +80,82 @@ var _ = ginkgo.Describe("Joining Process for aws flow", func() {
 			err = authn.ApproveSpokeClusterCSR(kubeClient, managedClusterName, time.Hour*24)
 			gomega.Expect(err).To(gomega.HaveOccurred())
 
-			// the hub kubeconfig secret should be filled after the ManagedCluster is accepted
-			// TODO: Revisit while implementing slice 3
-			//gomega.Eventually(func() error {
-			//	secret, err := util.GetFilledHubKubeConfigSecret(kubeClient, testNamespace, hubKubeconfigSecret)
-			//	if err != nil {
-			//		return err
-			//	}
-			//
-			//	// check if the proxyURL is set correctly
-			//	proxyURL, err := getProxyURLFromKubeconfigData(secret.Data["kubeconfig"])
-			//	if err != nil {
-			//		return err
-			//	}
-			//	if proxyURL != expectedProxyURL {
-			//		return fmt.Errorf("expected proxy url %q, but got %q", expectedProxyURL, proxyURL)
-			//	}
-			//	return nil
-			//}, eventuallyTimeout, eventuallyInterval).ShouldNot(gomega.HaveOccurred())
-
-			// the spoke cluster should have joined condition finally
-			// TODO: Revisit while implementing slice 3
-			//gomega.Eventually(func() error {
-			//	spokeCluster, err := util.GetManagedCluster(clusterClient, managedClusterName)
-			//	if err != nil {
-			//		return err
-			//	}
-			//	if !meta.IsStatusConditionTrue(spokeCluster.Status.Conditions, clusterv1.ManagedClusterConditionJoined) {
-			//		return fmt.Errorf("cluster should be joined")
-			//	}
-			//	return nil
-			//}, eventuallyTimeout, eventuallyInterval).ShouldNot(gomega.HaveOccurred())
+			// ensure that generated hub-kubeconfig-secret is correct
```

Comment:
let's also comment that the kubeconfig secret here in integration test for AWS won't be able to connect to hub server, since it is not in the eks environment. So we only validate the format of the hub-kubeconfig.